File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        int res = 0;
        int fq[26] = {0};

        for (int l = 0, r = 0; r < s.length(); r++) {
            fq[(s[r] & 31) - 1]++;

            while (fq[(s[r] & 31) - 1] > 2)
                fq[(s[l++] & 31) - 1]--;

            res = max(res, r - l + 1);
        }

        return res;
    }
};

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        res = l = 0
        fq = defaultdict(int)

        for r, ch in enumerate(s):
            fq[ch] += 1
            while fq[ch] > 2:
                fq[s[l]] -= 1
                l += 1
                
            res = max(res, r - l + 1)

        return res